### PR TITLE
improve type annotations in 'docutils.parsers.rst.directives.misc'

### PR DESCRIPTION
**Pull request by danieleades**
@JelleZijlstra i wasn't sure the best way to annotate the class variables in this one, since they are literal values (not changed at runtime) but can in principle be overridden by subclasses. What's the usual approach?

### CODE REVIEW

**Comment by danieleades** [comment]
pyright doesn't like the untyped base class

**Comment by danieleades** [comment]
this is blocked on https://github.com/python/typeshed/pull/11525, which is in turn blocked on https://github.com/python/typeshed/pull/11469

**Comment by JelleZijlstra** [comment]
> @JelleZijlstra i wasn't sure the best way to annotate the class variables in this one, since they are literal values (not changed at runtime) but can in principle be overridden by subclasses. What's the usual approach?

If a subclass can override them to any value of type `T` (where `T` is e.g. str or int), I'd allocate them as `ClassVar[T]`.

File: stubs/docutils/docutils/parsers/rst/directives/misc.pyi [hunk 1]
```diff
@@ -1,3 +1,35 @@
...
+
+class Class(Directive): ...
+
+class Role(Directive):
+    argument_pattern: Pattern[str]
+
+class DefaultRole(Directive): ...
+class Title(Directive): ...
+
+class MetaBody(SpecializedBody):
```
Comment (JelleZijlstra) [inline stubs/docutils/docutils/parsers/rst/directives/misc.pyi:28]:
```suggestion
# SpecializedBody has not yet been stubbed
class MetaBody(SpecializedBody):  # pyright: ignore[reportUntypedBaseClass]
```

**Pull request merged**